tx.go

```diff
@@ -93,6 +93,21 @@ func (tx *tx) CreateMapReduceJobs(stmt *influxql.SelectStatement, tagKeys []stri
 			}
 		}
 
+		// If a numerical aggregate is requested, ensure it is only performed on numeric data.
+		aggregates := stmt.FunctionCalls()
+		for _, a := range aggregates {
+			lit, ok := a.Args[0].(*influxql.VarRef)
+			if !ok {
+				return nil, fmt.Errorf("aggregate call didn't contain a field %s", a.String())
+			}
+			if influxql.IsNumeric(a) {
+				f := m.FieldByName(lit.Val)
+				if f.Type != influxql.Float || f.Type != influxql.Integer {
+					return nil, fmt.Errorf("aggregate '%s' requires numerical field values", a.Name)
```

Comment:
I would change this to `requires a numerical field. Field %s provided is of type %t`